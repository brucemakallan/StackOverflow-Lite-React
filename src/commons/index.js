const URL = 'https://stackoverflow-lite3-abm.herokuapp.com/api/v1';
export default URL;
