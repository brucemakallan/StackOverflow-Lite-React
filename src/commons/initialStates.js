const initialState = {
	signupReducer: {
		user: {
			fullName: '',
			email: '',
			password: '',
			retypePassword: ''
		}
	}
};

export const sampleUser = {
	fullName: 'Jack Katto',
	email: 'jack.katto@gmail.com',
	password: 'password#123',
	retypePassword: 'password#123'
};

export default initialState;
