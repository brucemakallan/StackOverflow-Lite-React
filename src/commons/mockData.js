const registrationMockData = {
	full_name: 'James Katto',
	email: 'jameskatto@gmail.com',
	password: 'password#123',
	retype_password: 'password#123'
};

export const loginMockData = {
	email: 'jameskatto@gmail.com',
	password: 'password#123'
};

export default registrationMockData;
