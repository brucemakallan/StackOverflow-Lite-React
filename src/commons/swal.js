export const swalError = {
	type: 'error',
	title: 'Oops...',
	text: 'Something went wrong!'
};

export const swalSuccess = {
	type: 'success',
	title: 'All good',
	text: 'Operation Successful'
};
