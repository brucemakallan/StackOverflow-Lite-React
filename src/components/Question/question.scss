$primaryDark: #0c337a;
$primaryLight: #007bff;
$accentColor: #a56b28;

.accent-color {
  color: $accentColor;
}
.primary-light {
  color: $primaryLight;
}
.icon {
  margin-right: 10px;
  cursor: pointer;
}
.green {
  color: #158d39;
}
.red {
  color: #d12937;
}

.question-card {
  margin: 15px;
  border: 1px solid #dddddd;
  border-radius: 5px;
  padding: 15px;

  .title {
    color: $primaryDark;
  }
}

.answer-card {
  margin: 10px 10px 10px 0;
  padding: 20px;
  background: #eeeeee;

  .answer-footer {
    display: flex;
    justify-content: space-between;
    padding-top: 10px;
    flex-flow: row wrap;

    .options {
      text-align: right;
    }
  }
}

.children {
  width: 50%;
}
@media all and (max-width: 800px) {
  .children {
    width: 100%;
    padding-top: 10px;
  }
  .answer-footer {
    .options {
      text-align: left !important;
    }
  }
}
