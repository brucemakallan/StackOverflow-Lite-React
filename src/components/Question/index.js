import React from 'react';
import PropTypes from 'prop-types';
import Answers from '../../containers/Answers';
import { library } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faQuestionCircle } from '@fortawesome/free-solid-svg-icons';
import { faClock } from '@fortawesome/free-regular-svg-icons';
import './question.scss';

library.add(faClock);
library.add(faQuestionCircle);

const Question = ({ question }) => (
	<div className="question-card">
		<h4 className="title">
			<FontAwesomeIcon className="primary-light icon" icon="question-circle" />
			{question.question}
		</h4>
		<div className="pl-2">
			<div className="date small mb-3">
				<FontAwesomeIcon className="icon" icon={['far', 'clock']} />
				{question.date_posted}
			</div>
			<h6 className="accent-color">Answers</h6>
			<div className="answers-box mb-2">
				<Answers questionId={question.id} questionTitle={question.question} />
			</div>
			<button type="button" className="btn btn-outline-primary btn-sm">Submit Answer</button>
		</div>
	</div>
);

Question.propTypes = {
	question: PropTypes.shape({}).isRequired
};

export default Question;
