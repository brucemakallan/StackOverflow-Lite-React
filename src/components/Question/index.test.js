import React from 'react';
import { shallow } from 'enzyme';
import Question from '.';
import { sampleQuestion } from '../../commons/initialStates';

describe('<Question /> component tests', () => {
	test('renders correctly', () => {
		const wrapper = shallow(<Question question={sampleQuestion} />);
		expect(wrapper).toMatchSnapshot();
	});
});
