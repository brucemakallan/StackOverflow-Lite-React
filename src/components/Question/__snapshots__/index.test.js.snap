// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Question /> component tests renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Question
    question={
      Object {
        "date_posted": "2018-12-27 17:51:48",
        "id": 1,
        "question": "What makes the world wonders so astounding?",
        "user_id": 1,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4
          className="title"
        >
          <FontAwesomeIcon
            border={false}
            className="primary-light icon"
            fixedWidth={false}
            flip={null}
            icon="question-circle"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
          What makes the world wonders so astounding?
        </h4>,
        <div
          className="pl-2"
        >
          <div
            className="date small mb-3"
          >
            <FontAwesomeIcon
              border={false}
              className="icon"
              fixedWidth={false}
              flip={null}
              icon={
                Array [
                  "far",
                  "clock",
                ]
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            2018-12-27 17:51:48
          </div>
          <h6
            className="accent-color"
          >
            Answers
          </h6>
          <div
            className="answers-box mb-2"
          >
            <Connect(Answers)
              questionId={1}
              questionTitle="What makes the world wonders so astounding?"
            />
          </div>
          <button
            className="btn btn-outline-primary btn-sm"
            type="button"
          >
            Submit Answer
          </button>
        </div>,
      ],
      "className": "question-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <FontAwesomeIcon
              border={false}
              className="primary-light icon"
              fixedWidth={false}
              flip={null}
              icon="question-circle"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />,
            "What makes the world wonders so astounding?",
          ],
          "className": "title",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "border": false,
              "className": "primary-light icon",
              "fixedWidth": false,
              "flip": null,
              "icon": "question-circle",
              "inverse": false,
              "listItem": false,
              "mask": null,
              "pull": null,
              "pulse": false,
              "rotation": null,
              "size": null,
              "spin": false,
              "symbol": false,
              "title": "",
              "transform": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "What makes the world wonders so astounding?",
        ],
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="date small mb-3"
            >
              <FontAwesomeIcon
                border={false}
                className="icon"
                fixedWidth={false}
                flip={null}
                icon={
                  Array [
                    "far",
                    "clock",
                  ]
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
              2018-12-27 17:51:48
            </div>,
            <h6
              className="accent-color"
            >
              Answers
            </h6>,
            <div
              className="answers-box mb-2"
            >
              <Connect(Answers)
                questionId={1}
                questionTitle="What makes the world wonders so astounding?"
              />
            </div>,
            <button
              className="btn btn-outline-primary btn-sm"
              type="button"
            >
              Submit Answer
            </button>,
          ],
          "className": "pl-2",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <FontAwesomeIcon
                  border={false}
                  className="icon"
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Array [
                      "far",
                      "clock",
                    ]
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />,
                "2018-12-27 17:51:48",
              ],
              "className": "date small mb-3",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "border": false,
                  "className": "icon",
                  "fixedWidth": false,
                  "flip": null,
                  "icon": Array [
                    "far",
                    "clock",
                  ],
                  "inverse": false,
                  "listItem": false,
                  "mask": null,
                  "pull": null,
                  "pulse": false,
                  "rotation": null,
                  "size": null,
                  "spin": false,
                  "symbol": false,
                  "title": "",
                  "transform": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "2018-12-27 17:51:48",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Answers",
              "className": "accent-color",
            },
            "ref": null,
            "rendered": "Answers",
            "type": "h6",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(Answers)
                questionId={1}
                questionTitle="What makes the world wonders so astounding?"
              />,
              "className": "answers-box mb-2",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "questionId": 1,
                "questionTitle": "What makes the world wonders so astounding?",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit Answer",
              "className": "btn btn-outline-primary btn-sm",
              "type": "button",
            },
            "ref": null,
            "rendered": "Submit Answer",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4
            className="title"
          >
            <FontAwesomeIcon
              border={false}
              className="primary-light icon"
              fixedWidth={false}
              flip={null}
              icon="question-circle"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            What makes the world wonders so astounding?
          </h4>,
          <div
            className="pl-2"
          >
            <div
              className="date small mb-3"
            >
              <FontAwesomeIcon
                border={false}
                className="icon"
                fixedWidth={false}
                flip={null}
                icon={
                  Array [
                    "far",
                    "clock",
                  ]
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
              2018-12-27 17:51:48
            </div>
            <h6
              className="accent-color"
            >
              Answers
            </h6>
            <div
              className="answers-box mb-2"
            >
              <Connect(Answers)
                questionId={1}
                questionTitle="What makes the world wonders so astounding?"
              />
            </div>
            <button
              className="btn btn-outline-primary btn-sm"
              type="button"
            >
              Submit Answer
            </button>
          </div>,
        ],
        "className": "question-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <FontAwesomeIcon
                border={false}
                className="primary-light icon"
                fixedWidth={false}
                flip={null}
                icon="question-circle"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />,
              "What makes the world wonders so astounding?",
            ],
            "className": "title",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "border": false,
                "className": "primary-light icon",
                "fixedWidth": false,
                "flip": null,
                "icon": "question-circle",
                "inverse": false,
                "listItem": false,
                "mask": null,
                "pull": null,
                "pulse": false,
                "rotation": null,
                "size": null,
                "spin": false,
                "symbol": false,
                "title": "",
                "transform": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "What makes the world wonders so astounding?",
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="date small mb-3"
              >
                <FontAwesomeIcon
                  border={false}
                  className="icon"
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Array [
                      "far",
                      "clock",
                    ]
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
                2018-12-27 17:51:48
              </div>,
              <h6
                className="accent-color"
              >
                Answers
              </h6>,
              <div
                className="answers-box mb-2"
              >
                <Connect(Answers)
                  questionId={1}
                  questionTitle="What makes the world wonders so astounding?"
                />
              </div>,
              <button
                className="btn btn-outline-primary btn-sm"
                type="button"
              >
                Submit Answer
              </button>,
            ],
            "className": "pl-2",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <FontAwesomeIcon
                    border={false}
                    className="icon"
                    fixedWidth={false}
                    flip={null}
                    icon={
                      Array [
                        "far",
                        "clock",
                      ]
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    title=""
                    transform={null}
                  />,
                  "2018-12-27 17:51:48",
                ],
                "className": "date small mb-3",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "border": false,
                    "className": "icon",
                    "fixedWidth": false,
                    "flip": null,
                    "icon": Array [
                      "far",
                      "clock",
                    ],
                    "inverse": false,
                    "listItem": false,
                    "mask": null,
                    "pull": null,
                    "pulse": false,
                    "rotation": null,
                    "size": null,
                    "spin": false,
                    "symbol": false,
                    "title": "",
                    "transform": null,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "2018-12-27 17:51:48",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Answers",
                "className": "accent-color",
              },
              "ref": null,
              "rendered": "Answers",
              "type": "h6",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(Answers)
                  questionId={1}
                  questionTitle="What makes the world wonders so astounding?"
                />,
                "className": "answers-box mb-2",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "questionId": 1,
                  "questionTitle": "What makes the world wonders so astounding?",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit Answer",
                "className": "btn btn-outline-primary btn-sm",
                "type": "button",
              },
              "ref": null,
              "rendered": "Submit Answer",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
